rust_src/src/fonts.rs

```diff
@@ -325,23 +336,25 @@ pub fn list_fonts(
     let mut frame: LispFrameRef = frame.into();
 
     let n = match num {
-        Some(n) if n < 0 => return Qnil,
+        Some(n) if n <= 0 => return Qnil,
         Some(n) => n,
         None => 0,
     } as usize;
 
     let list = unsafe { font_list_entities(frame.as_mut(), font_spec.into()) };
+
     match list.into() {
         Some((car, cdr)) if cdr.is_nil() => match car.as_vector() {
             Some(vec) if vec.len() == 1 => return list!(vec.get(0)),
             _ => (),
         },
-        _ => return Qnil,
+        Some(_) => (),
+        None => return Qnil,
     }
 
     let vec = match prefer {
-        None => unsafe { font_sort_entities(list, prefer.into(), frame.as_mut(), 0) },
-        Some(_) => vconcat_entity_vectors(list.into()),
+        None => vconcat_entity_vectors(list.into()),
+        Some(_) => unsafe { font_sort_entities(list, prefer.into(), frame.as_mut(), 0) },
```

Comment:
Would this version work?

`Some(font_spec)  => unsafe { font_sort_entities(list, font_spec.into(), frame.as_mut(), 0) },`

We already know that `prefer` is a `Some` the `into` call is going to perform these checks again.

Reply:
`font_spec` conflicts with already existing var. But otherwise seems to be working